### ISSUE TITLE
Updates of nested relations broken [was:Deutschland (country) is not a part of the see]

### ISSUE THREAD

**Issue by tomhughes** [tomhughes/trac-tickets#5093]
**[Submitted to the original trac issue database at 3.48pm, Wednesday, 8th January 2014]**

http://nominatim.openstreetmap.org/search.php?q=Deutschland&viewbox=

OSM: relation 1111111

**Comment by tomhughes** [comment]
**Author: lonvia**
**[Added to the original trac issue at 4.29pm, Wednesday, 8th January 2014]**

The little triangle is a remainder of [http://www.openstreetmap.org/api/0.6/relation/1111111/25 version 25] of the relation. I've removed it now but there is an interesting little update bug behind this which needs to be resolved. Title updated.

Technical details:
With one way in it, the relation used to be handled as a real multipolygon. When the way of version 25 was removed the relation became officially broken, which means that the old geometry of the relation is retained. osm2pgsql should somehow need to distinguish between a relation with a really broken geometry and a relation osm2pgsql does not handle (i.e. a nested relation).
